CRM/Batch/BAO/EntityBatch.php

```diff
@@ -23,9 +23,47 @@ class CRM_Batch_BAO_EntityBatch extends CRM_Batch_DAO_EntityBatch {
    * @return CRM_Batch_DAO_EntityBatch
    */
   public static function create($params) {
+    // Only write the EntityBatch record if the financial trxn and batch match on currency and payment instrument.
+    $batchCurrency = self::getBatchCurrency($params['batch_id']);
+    $batchPID = (int) CRM_Core_DAO::getFieldValue('CRM_Batch_DAO_Batch', $params['batch_id'], 'payment_instrument_id');
+    $trxn = \Civi\Api4\FinancialTrxn::get(FALSE)
+      ->addSelect('currency', 'payment_instrument_id')
+      ->addWhere('id', '=', $params['entity_id'])
+      ->execute()[0];
```

Comment:
If you change this from `[0]` to `->first()` at least one of the test fails will disappear (ie `  ->execute()->first()`)  `trxn` will be an empty array if nothing is returned - which may, or may not, cause e-notices lower down.

api\v4\Entity\ConformanceTest::testConformance with data set "EntityBatch" ('EntityBatch')
Undefined offset: 0